virtcontainers/factory/template/template.go

```diff
@@ -171,3 +178,19 @@ func (t *template) checkTemplateVM() error {
 	_, err = os.Stat(t.statePath + "/state")
 	return err
 }
+
+// Logger returns a logrus logger appropriate for logging template messages
+func (t *template) Logger() *logrus.Entry {
+	return templateLog.WithFields(logrus.Fields{
+		"subsystem": "template",
+	})
+}
+
+// SetLogger sets the logger for the factory template.
+func SetLogger(ctx context.Context, logger logrus.FieldLogger) {
+	fields := logrus.Fields{
+		"source": "virtcontainers",
```

Comment:
Nit: Shouldn't this be:

```go
"source": "virtcontainers/factory/template",
```

Reply:
I'm sorry, I  didn't understand it correctly. I have change it. @jodh-intel